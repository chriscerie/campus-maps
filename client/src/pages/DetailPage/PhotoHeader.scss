.detail-page-container {
  display: block;
  line-height: 2em;
}

.photo-header-container {
  position: relative;
  height: 25em;
  overflow: hidden;
  background: linear-gradient(transparent, black 110%);
}

.photo-header-content-container {
  color: #fff;
  box-sizing: border-box !important;
  align-items: flex-end;
  display: flex !important;
  position: absolute;
  padding: 2.5em 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  text-align: left;
}

.photo-header-content {
  display: inline-block;
  flex: 1 auto;
}

.photo-header-heading {
  font-size: 3em;
  font-weight: 700;
  line-height: 1.2em;
  letter-spacing: -0.4px;
}

.photo-header-subheading {
  font-size: 1.4em;
  font-weight: 400;
  line-height: 1.2em;
  letter-spacing: -0.4px;
}

#see-more-button {
  flex: 0 auto;
  border-radius: 0.3em;
  margin-top: -0.2em;
  border: 1px solid #ddd;
  padding: 0.5em 1.7em;
  text-decoration: none;
  font-weight: 700;
  transition: background-color 0.4s ease, color 0.4s ease;
  background-color: transparent;
  color: #fff;
}

#see-more-button:hover {
  transition: background-color 0.4s ease;
  background-color: rgba(255, 255, 255, 0.3);
}

#header-image {
  z-index: -1;
}
