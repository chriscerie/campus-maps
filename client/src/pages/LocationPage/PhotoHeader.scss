.location-page-container {
  display: block;
  line-height: 2em;
}

.photo-header-container {
  position: relative;
  height: 25em;
  overflow: hidden;
  background: linear-gradient(transparent, black 110%);
}

.photo-header-content-container {
  color: #fff;
  box-sizing: border-box !important;
  align-items: flex-end;
  display: flex !important;
  position: absolute;
  padding: 2.5em 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  text-align: left;
}

.photo-header-content {
  display: inline-block;
  flex: 1 auto;
}

.photo-header-heading {
  font-size: 3em;
  font-weight: 700;
  line-height: 1.2em;
  letter-spacing: -0.4px;
}

.photo-header-subheading {
  display: inline;
  font-size: 1.4em;
  font-weight: 400;
  line-height: 1.2em;
  letter-spacing: -0.4px;
  margin-right: 0.5em;
  color: #fff;
}

.photo-header-edit-button {
  @extend .photo-header-subheading;
  text-decoration: none;
  font-size: 0.9em;
  padding: 0.35em 0.7em;
  transition: background-color 0.8s ease;
  background-color: rgba(255, 255, 255, 0.25);
  border-radius: 0.3em;
  font-weight: 600;
}

.photo-header-edit-button:hover {
  transition: background-color 0.8s ease;
  background-color: rgba(255, 255, 255, 0.4);
}

#see-more-button {
  flex: 0 auto;
  border-radius: 0.3em;
  margin-top: -0.2em;
  border: 1px solid #ddd;
  padding: 0.5em 1.7em;
  text-decoration: none;
  font-weight: 700;
  transition: background-color 0.4s ease, color 0.4s ease;
  background-color: transparent;
  color: #fff;
}

#see-more-button:hover {
  transition: background-color 0.4s ease;
  background-color: rgba(255, 255, 255, 0.3);
}

#header-image {
  z-index: -1;
}
