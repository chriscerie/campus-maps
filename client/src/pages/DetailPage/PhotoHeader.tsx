import { Container, Stack } from '@mui/material';
import './PhotoHeader.scss';

function PhotoHeader() {
  return (
    <div className="detail-page-container">
      <div className="photo-header-container">
        <Container className="photo-header-content-container">
          <div className="photo-header-content">
            <div className="photo-header-heading">Phelps Hall</div>
            <div className="photo-header-subheading">University Building</div>
          </div>
          <a href="/loc" id="see-more-button">
            See more photos
          </a>
        </Container>
        <Stack
          direction="row"
          justifyContent="flex-start"
          alignItems="stretch"
          spacing={0}
          className="images-list-container"
        >
          <img
            src="https://www.engineering.ucsb.edu/sites/engineering.ucsb.edu/files/images/ucsbengineeringmap.jpg"
            id="header-image"
            alt="Map"
          ></img>
          <img
            src="https://www.engineering.ucsb.edu/sites/engineering.ucsb.edu/files/images/ucsbengineeringmap.jpg"
            id="header-image"
            alt="Map"
          ></img>
          <img
            src="https://www.engineering.ucsb.edu/sites/engineering.ucsb.edu/files/images/ucsbengineeringmap.jpg"
            id="header-image"
            alt="Map"
          ></img>
        </Stack>
      </div>
    </div>
  );
}

export default PhotoHeader;
