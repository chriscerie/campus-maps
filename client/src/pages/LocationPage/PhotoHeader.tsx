import { Container, Stack } from '@mui/material';
import { Link } from 'react-router-dom';
import './PhotoHeader.scss';
import CloseButton from '../../components/CloseButton';

function PhotoHeader() {
  return (
    <div className="location-page-container">
      <div className="photo-header-container">
        <Link to="/">
          <CloseButton
            sx={{
              position: 'absolute',
              right: 25,
              top: 25
            }}
          />
        </Link>
        <Container className="photo-header-content-container">
          <div className="photo-header-content">
            <div className="photo-header-heading">Phelps Hall
            </div>
            <div className="photo-header-subheading">University Building</div>
          </div>
          <Link to="/" id="see-more-button">
            See more photos
          </Link>
        </Container>
        <Stack
          direction="row"
          justifyContent="flex-start"
          alignItems="stretch"
          spacing={0}
          className="images-list-container"
        >
          <img
            src="https://www.engineering.ucsb.edu/sites/engineering.ucsb.edu/files/images/ucsbengineeringmap.jpg"
            id="header-image"
            alt="Map"
          ></img>
          <img
            src="https://www.engineering.ucsb.edu/sites/engineering.ucsb.edu/files/images/ucsbengineeringmap.jpg"
            id="header-image"
            alt="Map"
          ></img>
          <img
            src="https://www.engineering.ucsb.edu/sites/engineering.ucsb.edu/files/images/ucsbengineeringmap.jpg"
            id="header-image"
            alt="Map"
          ></img>
        </Stack>
      </div>
    </div>
  );
}

export default PhotoHeader;
