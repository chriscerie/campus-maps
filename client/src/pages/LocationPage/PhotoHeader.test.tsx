import ShallowRenderer from 'react-test-renderer/shallow';
import PhotoHeader from './PhotoHeader';

describe('PhotoHeader', () => {
  it('should render successfully', () => {
    const renderer = ShallowRenderer.createRenderer();
    const tree = renderer.render(<PhotoHeader name="name" type="type" />);
    expect(tree).toMatchSnapshot();
  });
});
