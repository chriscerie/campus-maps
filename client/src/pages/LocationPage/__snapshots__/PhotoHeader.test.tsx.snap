// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoHeader should render successfully 1`] = `
<div
  className="location-page-container"
>
  <div
    className="photo-header-container"
  >
    <ForwardRef(Container)
      className="photo-header-content-container"
    >
      <div
        className="photo-header-content"
      >
        <div
          className="photo-header-heading"
        >
          Phelps Hall
        </div>
        <div
          className="photo-header-subheading"
        >
          University Building
        </div>
      </div>
      <Link
        id="see-more-button"
        to="/"
      >
        See more photos
      </Link>
    </ForwardRef(Container)>
    <ForwardRef(Stack)
      alignItems="stretch"
      className="images-list-container"
      direction="row"
      justifyContent="flex-start"
      spacing={0}
    >
      <img
        alt="Map"
        id="header-image"
        src="https://www.engineering.ucsb.edu/sites/engineering.ucsb.edu/files/images/ucsbengineeringmap.jpg"
      />
      <img
        alt="Map"
        id="header-image"
        src="https://www.engineering.ucsb.edu/sites/engineering.ucsb.edu/files/images/ucsbengineeringmap.jpg"
      />
      <img
        alt="Map"
        id="header-image"
        src="https://www.engineering.ucsb.edu/sites/engineering.ucsb.edu/files/images/ucsbengineeringmap.jpg"
      />
    </ForwardRef(Stack)>
  </div>
</div>
`;
